BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -391,6 +409,43 @@ public final class BlueprintView: UIView {
         )
     }
 
+    /// Performs a full measurement and layout pass of all contained elements, and then collapses the nodes down
+    /// into `NativeViewNode`s, which represent only the view-backed elements. These view nodes
+    /// are then pushed into a `NativeViewController` to update the on-screen view hierarchy.
+    private func calculateNativeViewNodes(
+        in environment: Environment,
+        state: ElementStateTree
+    ) -> [(path: ElementPath, node: NativeViewNode)] {
+
+        guard let element = self.element else { return [] }
+
+        let node = calculateLayoutNodes(for: element, in: environment, state: state)
+
+        return node.resolve()
+    }
+
+    internal func calculateLayoutNodes(
+        for element: Element,
+        in environment: Environment,
+        state: ElementStateTree
+    ) -> LayoutResultNode {
+
+        rootState.root?.prepareForLayout()
+
+        defer {
+            self.rootState.root?.finishedLayout()
+        }
+
+        state.update(with: element, in: environment)
```

Comment:
Could we break this up into two functions, one where `element` is non-optional and returns a non-optional `root`, and another that takes an optional so we can avoid the force unwrapping?

Reply:
That's a good idea!

Reply:
Actually, this is a pain because it leads to ambiguous method resolution, having two methods with the same name, where one has `Element?` and another has `Element`; and giving them different names is a bit odd... Let's see if making the optional one just a `remove` works..

Reply:
That seems to work pretty well

Reply:
Would `@_disfavoredOverload` help?